tests/Operation/InsertManyFunctionalTest.php

```diff
@@ -200,4 +202,48 @@ public function testUnacknowledgedWriteConcernAccessesInsertedId(InsertManyResul
     {
         $this->assertInstanceOf(ObjectId::class, $result->getInsertedIds()[0]);
     }
+
+    public function testInsertingWithCodec(): void
+    {
+        $documents = [
+            TestObject::createForFixture(1),
+            TestObject::createForFixture(2),
+            TestObject::createForFixture(3),
+        ];
+
+        $expectedDocuments = [
+            (object) [
+                '_id' => 1,
```

Comment:
I've noted that TestDocumentCodec is ensuring the `$id` property gets mapped to `_id` here.